### DISCUSSION TITLE
Provide an example for dynamic routes for Navigation

### DISCUSSION THREAD

**Discussion by vince83110** [nuxt/nuxt#18065] [Questions]
Reading this documentation: https://v3.nuxtjs.org/docs/directory-structure/pages#dynamic-routes, I can't figure out how to use easily a NuxtLink to a dynamic component ?

I read at the official Vue Router doc too, but it's (for me) one the most important feature and the documentation deserves just one example of how to use dynamic route.

For now, I have: 

```
pages
|___product
    |___[slug].vue
```

And my link is:

```
<NuxtLink :to="{path: `/product/${product.seo.slug}`, component: Product}" class="product-view">
   .....
</NuxtLink>
```

It works, but the whole page is reloading, I don't have any transition effect and nothing in console.

How to use correctly ?

**Comment by lucpotage** [comment]
Hi @vince83110, I'm not sure to know the difference between dynamic routes and dynamic component.

For dynamic routes, you only have to write:

```vue
<NuxtLink :to="`/product/${product.seo.slug}`">...</NuxtLink>
```

Then, you put your `Product` component markup into the `[slug].vue` page component or import it from there.

**Reply by vince83110** [reply]
Yes I tried this way, I want to be sure because I am not a big fan of declaring the routes this way.

And I still didn't find a way to translate the routes ? do you know about it ? it was so easy on Nuxt 2.
